cpp/src/io/functions.cpp

```diff
@@ -313,79 +313,64 @@ table_with_metadata read_parquet(parquet_reader_options const& options,
   return reader->read(options);
 }
 
+/**
+ * @copydoc cudf::io::merge_rowgroup_metadata
+ */
+std::unique_ptr<std::vector<uint8_t>> merge_rowgroup_metadata(
+  const std::vector<std::unique_ptr<std::vector<uint8_t>>>& metadata_list)
+{
+  CUDF_FUNC_RANGE();
+  return detail_parquet::writer::merge_rowgroup_metadata(metadata_list);
+}
+
 // Freeform API wraps the detail writer class API
 std::unique_ptr<std::vector<uint8_t>> write_parquet(parquet_writer_options const& options,
                                                     rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
-  auto writer = make_writer<detail_parquet::writer>(options.get_sink(), options, mr);
+  namespace detail_pq = cudf::io::detail::parquet;
+
+  auto writer = make_writer<detail_parquet::writer>(
+    options.get_sink(), options, detail_pq::SingleWriteMode::YES, mr, rmm::cuda_stream_default);
 
   return writer->write(options.get_table(),
-                       options.get_metadata(),
                        options.is_enabled_return_filemetadata(),
-                       options.get_column_chunks_file_path(),
-                       options.get_decimal_precision());
+                       options.get_column_chunks_file_path());
 }
 
-/**
- * @copydoc cudf::io::merge_rowgroup_metadata
- */
-std::unique_ptr<std::vector<uint8_t>> merge_rowgroup_metadata(
-  const std::vector<std::unique_ptr<std::vector<uint8_t>>>& metadata_list)
+// Constructor to create parquet chunked writer
+parquet_chunked_writer::parquet_chunked_writer(chunked_parquet_writer_options const& op,
+                                               rmm::mr::device_memory_resource* mr)
 {
-  CUDF_FUNC_RANGE();
-  return detail_parquet::writer::merge_rowgroup_metadata(metadata_list);
+  namespace detail_pq = cudf::io::detail::parquet;
+  writer              = make_writer<detail_parquet::writer>(
+    op.get_sink(), op, detail_pq::SingleWriteMode::NO, mr, rmm::cuda_stream_default);
 }
 
-/**
- * @copydoc cudf::io::write_parquet_chunked_begin
- */
-std::shared_ptr<pq_chunked_state> write_parquet_chunked_begin(
-  chunked_parquet_writer_options const& op, rmm::mr::device_memory_resource* mr)
+// Moves writer unique pointer to object
+parquet_chunked_writer& parquet_chunked_writer::operator=(parquet_chunked_writer&& rhs)
```

Comment:
Isn't this what the default implementation would be? I feel like you don't need to declare this, it should be automatically generated.
AFAIK, since you have a `unique_ptr` member, copy constructor and assignment will not be generated, but the move ones should.

Reply:
I had an issue in cython earlier for which I added this, but now it seems not there, will remove this.